content/en/docs/concepts/security/security-checklist.md

```diff
@@ -0,0 +1,333 @@
+---
+reviewers:
+title: Security Checklist
+description: >
+  Baseline checklist for ensuring security in Kubernetes clusters.
+content_type: concept
+---
+
+<!-- overview -->
+
+This checklist aims at providing a basic list of guidance with links to more
+comprehensive documentation on each topic. It does not claim to be exhaustive
+and is meant to evolve.
+
+The order of topics does not reflect an order of priority.
+
+{{< caution >}}
+Checklists are **not** sufficient for attaining a good security posture on their
+own. A good security posture requires constant attention and improvement, but a
+checklist can be the first step on the never-ending journey towards security
+preparedness.  Some of the recommendations in this checklist may be too
+restrictive or too lax for your specific security needs. Since Kubernetes
+security is not "one size fits all", each category of checklist items should be
+evaluated on their merits.
+{{< /caution >}}
+
+<!-- body -->
+
+## Authentication & Authorization
+
+- [ ] `system:masters` group is not used for user or component authentication after bootstrapping.
+- [ ] The kube-controller-manager is running with `--use-service-account-credentials`
+  enabled.
+- [ ] The root certificate is protected (either an offline CA, or a managed
+  online CA with effective access controls).
+- [ ] Intermediate and leaf certificates have an expiry date no more than 3
+  years in the future.
+- [ ] A process exists for periodic access review, and reviews occur no more
+  than 24 months apart.
+
+See the [Role Based Access Control Good Practices](/docs/concepts/security/rbac-good-practices/)
+for guidance related to authentication and authorization.
+
+After bootstrapping, neither users nor components should authenticate to the
+Kubernetes API as `system:masters`. Similarly, running all of
+kube-controller-manager as `system:masters` should be avoided. In fact,
+`system:masters` should only be used as a break-glass mechanism, as opposed to
+an admin user.
+
+## Network security
+
+- [ ] CNI plugins in-use supports network policies.
+- [ ] Network policies to filter how workloads and services can communicate
+  inside of the cluster are enabled.
+- [ ] Use of a service mesh to encrypt all communications inside of the cluster.
+- [ ] The Kubernetes API is not exposed publicly on Internet.
```

Comment:
Do we want to mention Kubelet and etcd here as well? Just thinking that it's no unknown for people to put the kubelet on the Internet (IIRC shodan has ~232,000 kubelets) and there are some etcds too although fewer.

Reply:
Wow yes definitely, kubelet or etcd exposed on the internet is dramatic! Especially with the bad default of kubelet when not configured via a config file but only the flag.

Reply:
Tried to fix this here https://github.com/kubernetes/website/pull/33992/commits/cb1bcfcee07965b57127c907a1192d78abd9d07c.